### ISSUE TITLE
Use an export macro

### ISSUE THREAD

**Issue by donny-dont** [WebPlatformForEmbedded/libwpe#32]
Currently libwpe just uses `__attribute__((visibility("default")))` for exports. This doesn't work on all platforms such as windows.

**Comment by donny-dont** [comment]
@masali-hp should be hitting this issue as well with https://github.com/masali-hp/WPEBackend-rdk

**Comment by aperezdc** [comment]
The `visibility(…)` attribute is supported by GCC/Clang but not MSVC, which I suppose is a must for Windows support. Which means that @donny-dont is right and currently it's not possible to build on Windows with MSVC.

I see that one of the commits in the [masali-hp/WPEBackend-rdk](https://github.com/masali-hp/WPEBackend-rdk) repository uses this:

```c
#ifdef _MSC_VER
__declspec(dllexport)
#else
__attribute__((visibility("default")))
#endif
```

We could turn this into a `LIBWPE_EXPORT` macro in one of the headers, and use that to export the symbols. My experience with Windows development is limited, so if somebody with more knowledge can confirm whether this would be enough, that would be great (thanks in advance!).

**Comment by aperezdc** [comment]
This was implemented in #33 by @donny-dont (thanks!), so let's close it.
